From dbbd84c8d2ef9c2b9b22daefe586e0978f774176 Mon Sep 17 00:00:00 2001
From: Dan Smith <dansmith@redhat.com>
Date: Wed, 26 Jun 2024 08:41:02 -0700
Subject: [PATCH 7/7] Add safety check and detection support to FI tool

This adds a safety check and detection mechanism to the
tools/test_format_inspector.py utility for verifying those features
outside of glance.

Change-Id: I447e7e51315472f8fa6013d4c4852f54c1e0c43d
(cherry picked from commit 3a2528701afba3cbdf8c6485224d74bd547dd667)
(cherry picked from commit aba3561ee27f6ae8309e2fdfff7eb8cf0eb33b47)
(cherry picked from commit 23f621a02a22c25debbbe765c748295bab87f205)
---
 tools/test_format_inspector.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tools/test_format_inspector.py b/tools/test_format_inspector.py
index aa554386e..63e23210c 100755
--- a/tools/test_format_inspector.py
+++ b/tools/test_format_inspector.py
@@ -102,6 +102,13 @@ def main():
         else:
             print('Confirmed size with qemu-img')
 
+    print('Image safety check: %s' % (
+        fmt.safety_check() and 'passed' or 'FAILED'))
+    if args.input:
+        detected_fmt = format_inspector.detect_file_format(args.input)
+        print('Detected inspector for image as: %s' % (
+            detected_fmt.__class__.__name__))
+
 
 if __name__ == '__main__':
     sys.exit(main())
-- 
2.34.1

